app/posts/[url]/page.tsx

```diff
@@ -0,0 +1,34 @@
+import { allPosts } from '@/.contentlayer/generated';
+
+interface PageProps {
+  params: {
+    url: string;
+  };
+}
+
+export async function generateStaticParams(): Promise<PageProps['params'][]> {
```

Comment:
This function isn't doing any asynchronous stuff - what is the reason for making it `async`?

Reply:
The function doesn't currently have any asynchronous operations and I don't foresee the need for them in the future, really I must remove the async keyword and the Promise return type to simplify the code.